servicetalk-dns-discovery-netty/src/main/java/io/servicetalk/dns/discovery/netty/DefaultDnsServiceDiscovererBuilder.java

```diff
@@ -258,16 +261,23 @@ public DefaultDnsServiceDiscovererBuilder ndots(final int ndots) {
     }
 
     @Override
-    public DefaultDnsServiceDiscovererBuilder queryTimeout(final Duration queryTimeout) {
-        this.queryTimeout = queryTimeout;
+    public DefaultDnsServiceDiscovererBuilder queryTimeout(final @Nullable Duration queryTimeout) {
+        this.queryTimeout = queryTimeout == null ? null : DurationUtils.ensureNonNegative(queryTimeout, "queryTimeout");
+        return this;
+    }
+
+    @Override
+    public DefaultDnsServiceDiscovererBuilder resolutionTimeout(final @Nullable Duration resolutionTimeout) {
```

Comment:
IMO the name between `resolution` and `query` is quite close and not directly obvious how they are different. WDYT about using "total" in the name to make it more clear - i.e. `totalResolutionTimeout` in addition to the javadoc clarification you have in place?

Reply:
I debated between adding "total" or not in the name as well. Decided to drop it for 2 reasons:
1. Consistency with naming `DnsServiceDiscovererObserver.DnsResolutionObserver`. This timeouts starts when `onNewResolution` is called and stops when `DnsResolutionObserver` terminates.
2. Bcz of the SRV records, it's not really "total".